lib/rules/no-unused-prop-types.js

```diff
@@ -65,11 +65,10 @@ module.exports = {
      * @returns {Boolean} True if the prop is used, false if not.
      */
     function isPropUsed(node, prop) {
-      const usedPropTypes = node.usedPropTypes || [];
-      for (let i = 0, l = usedPropTypes.length; i < l; i++) {
-        const usedProp = usedPropTypes[i];
+      for (const usedProp of node.usedPropTypes || []) {
```

Comment:
i pretty intentionally didn't use `for..of` here, since the array iteration protocol is slow and brittle. i don't think this change is needed for this fix; let's revert it
```suggestion
      const usedPropTypes = node.usedPropTypes || [];
      for (let i = 0, l = usedPropTypes.length; i < l; i++) {
        const usedProp = usedPropTypes[i];
```